test_ffi/tests/bench.js

```diff
@@ -220,228 +221,288 @@ const dylib = Deno.dlopen(libPath, {
   },
 });
 
+const { nop } = dylib.symbols;
 Deno.bench("nop()", () => {
-  dylib.symbols.nop();
+  nop();
 });
 
+const { add_u32 } = dylib.symbols;
+Deno.bench("add_u32()", () => {
+  add_u32(1, 2);
+});
+
+const { nop_u8 } = dylib.symbols;
 Deno.bench("nop_u8()", () => {
-  dylib.symbols.nop_u8(100);
+  nop_u8(100);
 });
 
+const { nop_i8 } = dylib.symbols;
 Deno.bench("nop_i8()", () => {
-  dylib.symbols.nop_i8(100);
+  nop_i8(100);
 });
 
+const { nop_u16 } = dylib.symbols;
 Deno.bench("nop_u16()", () => {
-  dylib.symbols.nop_u16(100);
+  nop_u16(100);
 });
 
+const { nop_i16 } = dylib.symbols;
 Deno.bench("nop_i16()", () => {
-  dylib.symbols.nop_i16(100);
+  nop_i16(100);
 });
 
+const { nop_u32 } = dylib.symbols;
 Deno.bench("nop_u32()", () => {
-  dylib.symbols.nop_u32(100);
+  nop_u32(100);
 });
 
+const { nop_i32 } = dylib.symbols;
 Deno.bench("nop_i32()", () => {
-  dylib.symbols.nop_i32(100);
+  nop_i32(100);
 });
 
+const { nop_u64 } = dylib.symbols;
 Deno.bench("nop_u64()", () => {
-  dylib.symbols.nop_u64(100);
+  nop_u64(100);
 });
 
+const { nop_i64 } = dylib.symbols;
 Deno.bench("nop_i64()", () => {
-  dylib.symbols.nop_i64(100);
+  nop_i64(100);
 });
 
+const { nop_usize } = dylib.symbols;
 Deno.bench("nop_usize()", () => {
-  dylib.symbols.nop_usize(100);
+  nop_usize(100n);
 });
 
+const { nop_isize } = dylib.symbols;
 Deno.bench("nop_isize()", () => {
-  dylib.symbols.nop_isize(100);
+  nop_isize(100n);
 });
 
+const { nop_f32 } = dylib.symbols;
 Deno.bench("nop_f32()", () => {
-  dylib.symbols.nop_f32(100);
+  nop_f32(100.1);
 });
 
+const { nop_f64 } = dylib.symbols;
 Deno.bench("nop_f64()", () => {
-  dylib.symbols.nop_f64(100);
+  nop_f64(100.1);
 });
 
+const { nop_buffer } = dylib.symbols;
 const buffer = new Uint8Array(8).fill(5);
 Deno.bench("nop_buffer()", () => {
-  dylib.symbols.nop_buffer(buffer);
+  nop_buffer(buffer);
 });
 
+const { return_u8 } = dylib.symbols;
 Deno.bench("return_u8()", () => {
-  dylib.symbols.return_u8();
+  return_u8();
 });
 
+const { return_i8 } = dylib.symbols;
 Deno.bench("return_i8()", () => {
-  dylib.symbols.return_i8();
+  return_i8();
 });
 
+const { return_u16 } = dylib.symbols;
 Deno.bench("return_u16()", () => {
-  dylib.symbols.return_u16();
+  return_u16();
 });
 
+const { return_i16 } = dylib.symbols;
 Deno.bench("return_i16()", () => {
-  dylib.symbols.return_i16();
+  return_i16();
 });
 
+const { return_u32 } = dylib.symbols;
 Deno.bench("return_u32()", () => {
-  dylib.symbols.return_u32();
+  return_u32();
 });
 
+const { return_i32 } = dylib.symbols;
 Deno.bench("return_i32()", () => {
-  dylib.symbols.return_i32();
+  return_i32();
 });
 
+const { return_u64 } = dylib.symbols;
```

Comment:
This is somewhat problematic, if you don't do this destructure we will never get optimized call, right?

Reply:
Yes, but it's not really problematic. The prototype chain probably prevents function optimization, a good benchmark / performant code will do this.